### PR TITLE
fix [client/Mac]: ninja build

### PR DESCRIPTION
**Pull request by bmiklautz**
Ninja as cmake generator didn't build successfully on MacOS. With this modifications Ninja, Xcode and Unix Makefile can be used as generator and can build the MacOS client.

### CODE REVIEW

File: client/Mac/cli/CMakeLists.txt [hunk 1]
```diff
@@ -55,6 +55,7 @@ add_executable(${MODULE_NAME}
 	${${MODULE_PREFIX}_RESOURCES})
 
 set_target_properties(${MODULE_NAME} PROPERTIES OUTPUT_NAME "${MODULE_OUTPUT_NAME}") 
+add_dependencies(${MODULE_NAME} prepeare-framework-headers)
```
Comment (hardening) [inline client/Mac/cli/CMakeLists.txt:58]: typo ? `prepare-framework-headers` ?

File: client/Mac/CMakeLists.txt [hunk 2]
```diff
@@ -155,31 +165,41 @@ endif()
...
 	add_custom_command(TARGET ${MODULE_NAME} POST_BUILD
-		COMMAND ditto ${CMAKE_CURRENT_SOURCE_DIR}/${HEADER} ${CMAKE_CURRENT_BINARY_DIR}/$(CONFIGURATION)/${MODULE_OUTPUT_NAME}.framework/Headers/
+		COMMAND ditto ${CMAKE_CURRENT_SOURCE_DIR}/${HEADER} ${CMAKE_CURRENT_BINARY_DIR}/${safe_configuration}/${MODULE_OUTPUT_NAME}.framework/Headers/
+		${NinjaByproducts}
 		COMMENT Copying public header files to ${MODULE_NAME})
+	list(APPEND MAC_LIB_FRAMEWORK_HEADERS ${CMAKE_CURRENT_BINARY_DIR}/${CONFIGURATION}/${MODULE_OUTPUT_NAME}.framework/Headers/${HEADER})
 endforeach()
+add_custom_target(
+  prepeare-framework-headers
+  DEPENDS ${MAC_LIB_FRAMEWORK_HEADERS}
```
Comment (hardening) [inline client/Mac/CMakeLists.txt:196]: typo prepare-framework-headers  ?

File: client/Mac/CMakeLists.txt [hunk 3]
```diff
@@ -108,6 +108,16 @@ endforeach()
 
 set_property(TARGET ${MODULE_NAME} PROPERTY FOLDER "Client/Mac")
 
+if(CMAKE_GENERATOR MATCHES "Ninja")
+	if (NOT ${CONFIGURATION} STREQUAL "")
+		set(safe_configuration "$CONFIGURATION")
```
Comment (akallabeth) [inline client/Mac/CMakeLists.txt:113]: `$CONFIGURATION` and not `${CONFIGURATION}`?

File: client/Mac/CMakeLists.txt [hunk 4]
```diff
@@ -108,6 +108,16 @@ endforeach()
...
+	else()
+		set(safe_configuration "")
+	endif()
+else()
+	set(safe_configuration "$(CONFIGURATION)")
```
Comment (akallabeth) [inline client/Mac/CMakeLists.txt:118]: `$(CONFIGURATION)` and not `${CONFIGURATION}`?

File: client/Mac/CMakeLists.txt [hunk 5]
```diff
@@ -134,15 +144,15 @@ if (${BUILD_SHARED_LIBS})
...
+					"${CMAKE_CURRENT_BINARY_DIR}/${safe_configuration}/${MODULE_OUTPUT_NAME}.framework/Contents/$<TARGET_FILE_NAME:${DEST}>"
 				COMMENT Setting install name for ${LIB} in module ${DEST})
 		endforeach()
 	endforeach()
 
 endif()
 
 # Add post-build NIB file generation in unix makefiles. XCode handles this implicitly.
-# if("${CMAKE_GENERATOR}" MATCHES "Unix Makefiles")
+# if(NOT "${CMAKE_GENERATOR}" MATCHES "Xcode")
```
Comment (akallabeth) [inline client/Mac/CMakeLists.txt:155]: changes in a commented line?

File: client/Mac/cli/CMakeLists.txt [hunk 6]
```diff
@@ -66,9 +67,19 @@ set_target_properties(${MODULE_NAME} PROPERTIES RESOURCE "${${MODULE_PREFIX}_RES
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -F../")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -F../")
 
+if(CMAKE_GENERATOR MATCHES "Ninja")
```
Comment (akallabeth) [inline client/Mac/cli/CMakeLists.txt:70]: this looks like some duplication? you already defined that one directory above, so it should be defined here? (if not move the definition before the `add_subdirectory`)

Comment (akallabeth) [reply client/Mac/cli/CMakeLists.txt:70]: also, same notes as above apply

**Review by akallabeth — CHANGES_REQUESTED** [review]
I can not compile on mac with this:
```
FAILED: client/Mac/cli/CMakeFiles/MacFreeRDP.dir/AppDelegate.m.o 
/opt/local/bin/ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -DFREERDP_EXPORTS -DWITH_OPENSSL -DWITH_VERBOSE_WINPR_ASSERT -I/opt/local/libexec/openssl3/include -I/Users/dev/src/freerdp/winpr/include -I/Users/dev/src/freerdp/ninja/winpr/include -I/Users/dev/src/freerdp/ninja -I/Users/dev/src/freerdp/ninja/include -I/Users/dev/src/freerdp/include -F../ -O3 -DNDEBUG -flto=thin -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -mmacosx-version-min=10.15 -fPIE -std=gnu11 -MD -MT client/Mac/cli/CMakeFiles/MacFreeRDP.dir/AppDelegate.m.o -MF client/Mac/cli/CMakeFiles/MacFreeRDP.dir/AppDelegate.m.o.d -o client/Mac/cli/CMakeFiles/MacFreeRDP.dir/AppDelegate.m.o -c /Users/dev/src/freerdp/client/Mac/cli/AppDelegate.m
In file included from /Users/dev/src/freerdp/client/Mac/cli/AppDelegate.m:9:
/Users/dev/src/freerdp/client/Mac/cli/AppDelegate.h:10:9: fatal error: 'MacFreeRDP/MRDPView.h' file not found
#import <MacFreeRDP/MRDPView.h>
        ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
```

File: client/Mac/CMakeLists.txt (hunk 5 above)
Comment (bmiklautz) [reply client/Mac/CMakeLists.txt:155]: Yes. Adapt the code, if locally used or for whatever reasons, to apply to *all* generators that are not Xcode and not only to 'Unix Makefiles'

File: client/Mac/CMakeLists.txt (hunk 2 above)
Comment (bmiklautz) [reply client/Mac/CMakeLists.txt:196]: good catch!

File: client/Mac/CMakeLists.txt (hunk 4 above)
Comment (bmiklautz) [reply client/Mac/CMakeLists.txt:118]: Unfortunately it's not that obvious with the configuration. How it was created $(CONFIGURATION) is used directly in the generator when a custom command is invoked. That's why it's $(CONFIGURATION) for make and $CONFIGURATION for ninja.

File: client/Mac/cli/CMakeLists.txt (hunk 6 above)
Comment (bmiklautz) [reply client/Mac/cli/CMakeLists.txt:70]: I'd generally agree but as client/Mac and client/Mac/cli are created as separate cmake `project`s and both affected by the same issue I did copy it as a band aid fix. In general the MacOS client building could use "some" improvements..

**Comment by akallabeth** [comment]
@freerdp-bot test

**Review by akallabeth — APPROVED** [review]
LGTM,
works with latest `cmake`

**Pull request merged**